### ISSUE TITLE
Default properties in Oracle database are not modified

### ISSUE THREAD

**Issue by noeliajimenezg** [micronaut-projects/micronaut-test-resources#663]
### Expected Behavior

The container should start with this custom configuration
```
test-resources.containers.oracle.db-name=MYDB
test-resources.containers.oracle.username=micronaut
test-resources.containers.oracle.password=micronaut
datasources.default.db-name=MYDB
```

### Actual Behaviour

The container starts with the default configuration.
The code passes by
1. https://github.com/micronaut-projects/micronaut-test-resources/blob/3aeed97fd90e873186eac60477f55b5b8f3743d8/test-resources-jdbc/test-resources-jdbc-oracle-free/src/main/java/io/micronaut/testresources/oracle/free/OracleFreeTestResourceProvider.java#L80
2. https://github.com/micronaut-projects/micronaut-test-resources/blob/3aeed97fd90e873186eac60477f55b5b8f3743d8/test-resources-jdbc/test-resources-jdbc-core/src/main/java/io/micronaut/testresources/jdbc/AbstractJdbcTestResourceProvider.java#L120

Debugging, I saw that in method "ifPresent", it tries to extract the wrong prefix. Same result with .properties format or .yml.

![Capture d’écran 2024-07-18 à 03 00 05](https://github.com/user-attachments/assets/c9423b29-c99d-4c17-b1ff-ed964e1237b9)

### Steps To Reproduce

Run the example project
I tried using the gradle plugin or directly adding the libraries but the result is the same.

### Environment Information

- Mac OS
- JDK 21
- test-resources 2.5.3

### Example Application

https://github.com/noeliajimenezg/micronaut_error_testresources

### Version

4.5.0

**Comment by melix** [comment]
So I think the issue is actually quite different. First of all, your `application-test` file should be `.properties`, not `.yaml`. This alone explains the original issue. Next, you shouldn't add test resources dependencies to the test classpath : it runs in a separate process. Here's an updated build script:

```gradle
plugins {
    id("com.github.johnrengelman.shadow") version "8.1.1"
    id("io.micronaut.application") version "4.4.0"
    id("io.micronaut.test-resources") version "4.4.0"
}

version = "0.1"
group = "com.example"

repositories {
    mavenCentral()
}

configurations.configureEach {
    resolutionStrategy.eachDependency { details ->
        if (details.requested.group == 'io.micronaut.testresources') {
            details.useVersion "2.5.3"
        }
    }
}

dependencies {
    annotationProcessor("io.micronaut:micronaut-http-validation")
    annotationProcessor("io.micronaut.serde:micronaut-serde-processor")
    implementation("io.micronaut.serde:micronaut-serde-jackson")
    compileOnly("io.micronaut:micronaut-http-client")
    runtimeOnly("ch.qos.logback:logback-classic")
    testImplementation("io.micronaut:micronaut-http-client")
    testImplementation("io.micronaut.testresources:micronaut-test-resources-extensions-core")
    testImplementation("io.micronaut.testresources:micronaut-test-resources-extensions-junit-platform")


//  jdbc dependencies: in our case the real connection is made by a custom jdbc library
//  micronaut-data-jdbc triggers the addition of jdbc-oracle-free, only in runtimeOnly
//    runtimeOnly 'io.micronaut.data:micronaut-data-jdbc:4.8.3'
    testRuntimeOnly('io.micronaut.sql:micronaut-jdbc-hikari')
    testRuntimeOnly("com.oracle.database.jdbc:ojdbc11")
    testRuntimeOnly("org.flywaydb:flyway-database-oracle")
    testImplementation("io.micronaut.flyway:micronaut-flyway")
}


application {
    mainClass.set("com.example.Application")
}
java {
    sourceCompatibility = JavaVersion.toVersion("21")
    targetCompatibility = JavaVersion.toVersion("21")
}


graalvmNative.toolchainDetection = false
def testResourcesEnabled = providers.gradleProperty("micronaut.testResources").map(Boolean::parseBoolean).orElse(false)
micronaut {
    runtime("netty")
    testRuntime("junit5")
    processing {
        incremental(true)
        annotations("com.example.*")
    }
    testResources {
        version = "2.5.3"
        enabled = testResourcesEnabled
        clientTimeout = 500 //to avoid timeouts first time pulling oracle image
        additionalModules.add('jdbc-oracle-free') //this triggers the database if micronaut-data is in runtime classpath not test runtime classpath
    }
}

test {
    systemProperty 'micronaut.env.deduction', 'false'
    if (testResourcesEnabled.get()){
        systemProperty 'micronaut.environments', 'test'
    }
    systemProperty 'micronaut.test.resources.enabled', testResourcesEnabled.get()
    useJUnitPlatform()
}

tasks.named("dockerfile") {
    baseImage("eclipse-temurin:21-jre-jammy")
}

tasks.named("dockerfileNative") {
    jdkVersion = "21"
}
```

**Comment by noeliajimenezg** [comment]
Thanks @melix
Yes, my initial configuration is a .properties and by using the plugin with additionalModules.add('jdbc-oracle-free') and just adding the extensions.
However, if you want to debug on IntelliJ the test resource service, the plugin is not enough.
Unfortunately, your modifications on build.gradle does not change the result.

Were you able to set a custom database name, username and password?

**Comment by melix** [comment]
Yes, you are correct that there's something fishy since what the server gets is a map with `test-resources` prefixes, which shouldn't happen. I'm investigating.

**Comment by melix** [comment]
Fixed in 2.5.4
